### DISCUSSION TITLE
有没有脚本可以实现真正的历史记录

### DISCUSSION THREAD

**Discussion by Youxikong** [hooke007/mpv_PlayKit#197] [Q&A 问答] [answered]
我用过mpv.net一段时间，这个软件有个历史记录功能，而且关闭后再次打卡，可以直接播放上一个视频。换到mpv后，mpv本身没有历史记录功能，找了相关脚本，用起来也是类似一个额外的播放列表，并没有打开后直接播放上一个视频的效果。请问这个需要该如何能实现吗。

**Comment by hooke007** [comment, marked as answer]
<s>mpvnet打开后直接播放旧视频？我怎么不记得有这个功能.你说的是那个右键里的历史记录吗</s>
我知道你在说哪个命令了，其实也就是相当于省略了几个点点点的步骤

从功能上讲 https://github.com/Eisa01/mpv-scripts#simplehistor 差不多

**Reply by Youxikong** [reply]
我希望的是，打开点击播放，就直接比方最后一个视频了。现在键盘鼠标点点点比较麻烦。

**Reply by hooke007** [reply]
这种行为需要脚本里提供相应函数和可供外部绑定的命令。

**Reply by dyphire** [reply]
你说的是mpvnet空闲下的空格键打开最后播放文件的操作？
上面提到的SimpleHistory.lua默认绑定的键位ctrl+r也实现了相同的功能。嫌动作多的话也可以自己绑到空格键上，效果是一样的

**Reply by Youxikong** [reply]
这也是一种方法。键位冲突又是另一个问题了。

**Reply by yeezylife** [reply]
你实际用一下这个脚本就知道了。打开mpv程序，他会默认加载recent history列表，这个时候你只需要按一下回车（或者按照你的习惯设置成空格），就是播放上次播放的视频了。

开始播放后历史列表自动关闭，此时空格键的功能变回了 播放/暂停。不存在冲突。

这个脚本很好用 win11只能自动记录我打开的第一个文件，于是往往page down往下看几集之后，一退出就不知道自己看到哪一集了，解决了我的一个痛点。。
